homeassistant/components/vizio/media_player.py

```diff
@@ -274,6 +286,16 @@ async def _async_update_options(self, config_entry: ConfigEntry) -> None:
         self._volume_step = config_entry.options[CONF_VOLUME_STEP]
         self._conf_apps.update(config_entry.options.get(CONF_APPS, {}))
 
+    async def async_update_setting(
+        self, setting_type: str, setting_name: str, new_value: Union[int, str]
+    ) -> None:
+        """Update a setting when update_setting service is called."""
+        await self._device.set_setting(
+            setting_type.lower().replace(" ", "_"),
```

Comment:
The value modification should be moved to the voluptuous service schema.

Reply:
Fair, I will include this in the PR